orc8r/cloud/go/services/bootstrapper/servicers/registration/storage.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"github.com/pkg/errors"
+
+	"magma/orc8r/cloud/go/blobstore"
+	"magma/orc8r/cloud/go/services/bootstrapper"
+	"magma/orc8r/cloud/go/storage"
+	merrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/protos"
+)
+
+const networkWildcard = "*"
+
+type Store interface {
+	SetTokenInfo(tokenInfo protos.TokenInfo) error
+	GetTokenInfoFromLogicalID(networkID string, logicalID string) (*protos.TokenInfo, error)
+	GetTokenInfoFromNonce(nonce string) (*protos.TokenInfo, error)
+}
+
+type blobstoreStore struct {
+	factory blobstore.StoreFactory
+}
+
+func NewBlobstoreStore(factory blobstore.StoreFactory) Store {
+	return &blobstoreStore{factory}
+}
+
+func (b *blobstoreStore) SetTokenInfo(tokenInfo protos.TokenInfo) error {
+	isUnique, err := b.isNonceUnique(tokenInfo.Nonce)
```

Comment:
We want this check to occur within the same tx -- so after the "start transaction"

Reply:
`IsNonceUnique` calls `GetTokenInfoFromNonce` which has its own transaction. Would you prefer for me to pull out the part inside of the transaction from `GetTokenInfoFromNonce`? Or leave this as is

Reply:
If we're using txs, the check needs to be in the same tx as the action. So whichever impl you think works best for that